integration-tests/public/rlp-cross-contract/lib.rs

```diff
@@ -0,0 +1,58 @@
+//! A smart contract to test using RLP encoding for EVM compatibility.
+
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[ink::contract(abi_encoding = "rlp")]
+mod rlp_cross_contract {
+    use ink::env::call::{build_call, ExecutionInput};
+    use ink::env::{debug_println, CallFlags};
+    use crate::keccak_selector;
+
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct RlpCrossContract {}
+
+    impl RlpCrossContract {
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self { }
+        }
+
+        #[ink(message)]
+        pub fn call_contract_rlp(&mut self, callee: [u8;32]) {
+            let selector = keccak_selector(b"flip");
+            let callee: AccountId = callee.into();
+
+            // TODO: fails due to call builder encoding with scale
```

Comment:
more accurately, issue is coming from here: https://github.com/use-ink/ink/blob/ea20e21ab37a6bc57f096050316f6638c75bc047/crates/env/src/engine/on_chain/impls/pallet_contracts.rs#L530